File: python progs/stopwatch.py
# Command given to ChatGPT ChatGPT (chat generative pre-trained transformer)...
# write a stopwatch program in python

import time

# define the start time
start = time.time()

# keep looping until user presses 'q'
while True:
    # print the elapsed time
    print('Elapsed Time: {:.2f}s'.format(time.time() - start))
    
    # prompt user to quit
    user_input = input('Press "q" to quit: ')
    
    if user_input == 'q':
        break
        
"""
ian@hp:~/openai-quickstart-node/notes/python progs$ python stopwatch.py
Elapsed Time: 0.00s
Press "q" to quit: #  <--- pressed return key after 30 seconds
Elapsed Time: 32.53s
Press "q" to quit: 
Elapsed Time: 35.73s

"""

File: python progs/stopwatch.cpp
/* Command given to ChatGPT ChatGPT (chat generative pre-trained transformer)...
write a stopwatch program for a computer 
*/

#include <iostream>
#include <chrono>


int main()
{
    std::cout << "Press any key to start the stopwatch.\n";
    std::cin.get();
    auto startTime = std::chrono::steady_clock::now();

    std::cout << "Press any key to stop the stopwatch.\n";
    std::cin.get();
    auto endTime = std::chrono::steady_clock::now();

    auto elapsedTime = endTime - startTime;
    auto elapsedTimeInSeconds = std::chrono::duration_cast<std::chrono::seconds>(elapsedTime).count();

    std::cout << "Elapsed time: " << elapsedTimeInSeconds << " seconds\n";

    return 0;
}

/*
This is a c++ program use g++ compiler...

$ g++ stopwatch.cpp -o stopwatch

$ ./stopwatch
Press any key to start the stopwatch.

Press any key to stop the stopwatch.

Elapsed time: 4 seconds

*/
